provider/sgx.py

```diff
@@ -0,0 +1,200 @@
+    def verify_sgx_flags(self, qmp_command, flags):
+        """
+        Check if sgx cpu flags enabled in qmp cmd
+
+        :param qmp_command: query sgx qmp command output
+        :param flags: sgx flags need to be verified
+        """
+        for ele in flags:
+            if qmp_command[ele] is not True:
+                self._test.fail("%s is not enabled, qmp check failed." % ele)
+
+    def get_config_total_epc_size(self):
+        """
+        Get total epc size from configuration.
+
+        :return: total configured epc size
+        """
+        epc_list = self._params["vm_sgx_epc_devs"].split()
+        config_epc_size = 0
+        for ele in epc_list:
+            epc_params = self._params.object_params(ele)
+            ele_memdev = self._params.object_params(
+                epc_params["vm_sgx_epc_memdev"])
+            tmp_epc_size = int(float(normalize_data_size(
+                ele_memdev["size_mem"], "B", 1024)))
+            config_epc_size += tmp_epc_size
+        return config_epc_size
+
+    def get_config_epc_numa_info(self):
+        """
+        Get epc size on each nume node from configuration.
+
+        :return: epc size on each nume node configured.
+        """
+        guest_sgx_epc_list = self._params["vm_sgx_epc_devs"].split()
+        tmp_epc_dict = {}
+        for ele in guest_sgx_epc_list:
+            epc_params = self._params.object_params(ele)
+            ele_memdev = self._params.object_params(
+                epc_params["vm_sgx_epc_memdev"])
+            tmp_epc_size = int(float(normalize_data_size(
+                ele_memdev["size_mem"], "B", 1024)))
+            epc_numa_id = int(epc_params["vm_sgx_epc_node"])
+            tmp_epc_dict = {'size': tmp_epc_size, 'node': epc_numa_id}
+        return tmp_epc_dict
+
+    def verify_qmp_host_sgx_cap(self):
+        """
+        Verify query host sgx capabilities qmp cmd in sgx flags and epc size
+        """
+        sgx_flags = self._params['sgx_flags'].split()
+        host_sgx_info = self._monitor.query_sgx_capabilities()
+        self.verify_sgx_flags(host_sgx_info, sgx_flags)
+
+        host_cap = SGXHostCapability(self._test, self._params)
+        host_epc_size = host_cap.host_epc_size
+        host_qmp_section_size = host_sgx_info['section-size']
+        if host_epc_size != int(host_qmp_section_size):
+            self._test.fail("Host epc size %s is not equal to query sgx"
+                            "capabilities section size %s"
+                            % (host_epc_size, host_qmp_section_size))
+
+    def verify_qmp_guest_sgx_cap(self):
+        """
+        Verify query guest sgx capabilities qmp cmd in sgx flags and epc size
+        """
+        sgx_flags = self._params['sgx_flags'].split()
+        guest_sgx_info = self._monitor.query_sgx()
+        self.verify_sgx_flags(guest_sgx_info, sgx_flags)
+        LOG_JOB.debug("Guest query SGX flags %s verified done", sgx_flags)
+
+        epc_sections_info = guest_sgx_info['sections']
```

Comment:
@zhencliu , as you suggested, I add a new function get_config_epc_numa_info. This and get_config_total_epc_size will get total epc size and epc size on each numa from cfg.